pkg/genericcache/genericcache.go

```diff
@@ -0,0 +1,89 @@
+package genericcache
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/portworx/sched-ops/k8s/apiextensions"
+	"github.com/sirupsen/logrus"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+type genericCache struct {
+	crdKindToPluralMap map[string]string
+}
+
+// GenericCache is a generic structure which will implement mostly non-k8s resources related information.
+// Individual members get and set need to be provided by the interface.
+type GenericCache interface {
+	// GetCRDKindToPluralMap returns the crdKindToPluralMap from generic cache singleton.
+	GetCRDKindToPluralMap() map[string]string
+
+	// SetCRDKindToPluralMap will set a new CRD's plural in the map
+	SetCRDKindToPluralMap(kind, plural string)
```

Comment:
Lets rename this to `SetPluralForCRDKind`. The associated Get function returns an actual map, but this function doesn't take a map as an input.

Reply:
Done